src/pages/ResultPage/ResultPage.tsx

```diff
@@ -0,0 +1,25 @@
+import { useLocation, useNavigate } from "react-router-dom";
+
+const ResultPage = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const isFailed = location.state?.isFailed;
+  const mistakesNumber = location.state?.mistakesNumber;
+
+  const onRetry = () => navigate("/");
+
+  return isFailed ? (
+    <div className="failed">
+      <p>Ooops... you failed, you did more than {mistakesNumber} mistake(s)</p>
+      <button onClick={onRetry}>pass the test again</button>
+    </div>
+  ) : (
+    <div className="result">
+      <p>Great news! You passed the test!</p>
+      <button onClick={onRetry}>pass the test again</button>
```

Comment:
You have 2 same buttons (line 15 and 20) with the same text and behavior. It is better to have only 1 button (or Link instead)

Reply:
fixed